---
title: "MICB425"
author: "Zhong (Zack) Dang 22481148"
date: "January 21, 2018"
output:
  html_document: default
  pdf_document: default
---
###Module 1 Portfolio Check

* Installation check
    + Completion status:
    + Comments:
* Portfolio repo setup
    + Completion status: 
    + Comments:
* RMarkdown Pretty html Challenge
    + Completion status:
    + Comments:
* Evidence worksheet_01
    + Completion status:
    + Comments:
* Evidence worksheet_02
    + Completion status:
    + Comments:
* Evidence worksheet_03
    + Completion status:
    + Comments:
* Problem Set_01
    + Completion status:
    + Comments:
* Problem Set_02
    + Completion status:
    + Comments:
* Writing assessment_01
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments: 

#Module 1

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Evidence Worksheet_01 for "Prokaryotes: The Unseen Majority"

### Learning Objectives
Describe the numerical abundance of microbial life in relation to ecology and biogeochemistry of Earth systems.

### General Questions
#### What were the main questions being asked?

This article asks and attempts to answer the following questions:

1. What is the total number of prokaryotic cells in the 3 main habitats and the cumulative total on the planet?

2. What is the total amount of carbon contained in prokaryotic cells in the 3 main habitats and the global total?

3. What is the average turnover rate and productivity of prokaryotes in the 3 main habitats and the global total?

#### What were the primary methodological approaches used?

Cell density data from samples ofthe 3 main representative habitats (Aquatic, soil and subsurface environments) were analyzed and total number estimations were generated. Aquatic numbers were estimated from mean sample values of several primary studies. Soil numbers were estimated based on cultivated soil sample numbers.Subsurface numbers were extrapolated from various studies' sample data, but also estimated from calculating average porosity of terrestrial subsurface environments and average subsurface prokaryotic volume, and also from groundwater sample data. Carbon content was estimated in soil and subsurface environments as half the dry weight of the average prokaryotic cell. In aquatic environments, the upper estimate of carbon per cell is used. 

#### Summarize the main results or findings. 

The total number of prokaryotes on Earth is estimated to be 4-6x10^30 cells and the total amount of prokarotic carbon is esitmated to be 350-550 Pg, which is 60-100% of the total carbon in plants. Prokaryotes also cumulativey contain 10 times the amount of phosphorous and nitrogen than plants. Heterotropic prokaryote turnover is fastest in the upper 200m of the ocean and slowest in subsurface environments. The estimated global cellular production rate is 1.7x10^30 cells/yr. This also leads to a large capacity for genetic diversification through mutation.

#### Do new questions arise from the results?

1. How can subsurface estimation methods be refined to more accurately characterize the subsurface prokaryotic world?

2. How can phylogenetic analysis methods be changed to account for the high degree of diversity and mutation rate in prokaryotes?

3. With the differences in prokaryotic genomes and evolution through mutation, how do we definte a prokaryotic species?

#### Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 

Overall, the paper did a decent job of combining sample data from multiple papers and generating population estimates of the 3 major habitats and representing this data in easy-to-read tables. However, much of this data is based on estimation, with the extrapolation of subsurface numbers having a high degree of error. Some assumptions made include the use of an average prokaryotic cell volume and an average carbon content per cell value, on which global total estimates are made. Despite this, the evidence is adequate given our current models and techniques and thus the conclusions are valid. The paper rightfully acknowledges the limitations of some methods and discrepancies between studies. With the development of sampling methods and improved metagenomic analytical techniques, more accurate estimations can be made in the future. 

## Evidence Worksheet_02 â€œLife and the Evolution of Earthâ€™s Atmosphereâ€

### Learning objectives: 
Comment on the emergence of microbial life and the evolution of Earth systems
    
#### Indicate the key events in the evolution of Earth systems at each approximate moment in the time series. If times need to be adjusted or added to the timeline to fully account for the development of Earth systems, please do so. 

    + 4.6 billion years ago - Formation of the Solar System from a local accretion disk, creating the Sun and the planets.
    
    + 4.2 billion years ago - Formation of the oceans, creating the land-sea surface we know today. There is some evidence that plate tectonics started at this time, as well as the first amino acids and RNA.
    
    + 3.8 billion years ago - Earliest evidence of life in the form of cells.
    
    + 3.75 billion years ago - A group splits off from the last common ancestor and forms the domain, Archaea.
    
    + 3.5 billion years ago - Evolution of the domain, Prokaryotes and evolution of photosynthesis. There is fossil evidence for the formation of microbial aggregations and biofilms. 
    
    + 3.0 billion years ago - First evidence for the presence of viruses
    
    + 2.7 billion years ago - Evolution of cyanobacteria
    
    + 2.2 billion years ago - The Great Oxygenation Event, brought upon by massed photosynthesis by cyanobacteria. 
    
    + 2.1 billion years ago - Evolution of Eukarya and later (2.0 Gya), evidence of endosymbiosis by Eukarya to form chloroplasts and mitochondria
    
    + 1.3 billion years ago - Lineages that eventually formed plant, animal and fungal kingdoms split off from the main eukaryote ancestral line
    
    + 550,000 years ago - First land plants evolved
    
    + 200,000 years ago - Evolution of mammals


#### Describe the dominant physical and chemical characteristics of Earth systems at the following waypoints:  

    + Hadean - The early Hadean might have been characterized by global glaciation due to a weak, young Sun. Later, a very hot atmosphere comprised mainly of water vapour and CO2 formed, with a silica crust on the surface
    
    + Archean - About 3 times hotter than the current Earth, with a liquid water acidic ocean and a mostly CO2 atmosphere
    
    + Precambrian - The evolution of photosynthesis and the massed expansion of cyanobacteria caused a huge increase in atmospheric oxygen, called the Great Oxygenation Event
    
    + Proterozoic  - Oxygen rich, hot atmosphere
    
    + Phanerozoic  - Exolution of land plants contributes to a further increase in atmospheric oxygen levels. The Earth is also slightly cooler. 

## Evidence Worksheet_03 "The Anthropocene"

### Learning Objectives
Evaluate human impacts on the ecology and biogeochemistry of Earth systems.

### General Questions
#### What were the main questions being asked?

1. When did the Anthropocene begin?

2. Have humans created imbalances in modern biogeochmical cycles (particularly the N and C cycles), and how would this affect the cycles of the future?

3. How will the engineered Anthropocene affect global biodiversity and the climate?

#### What were the primary methodological approaches used?



#### Summarize the main results or findings. 



#### Do new questions arise from the results?



#### Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 



## Problem set_01

## Problem set_02 ??Microbial Engines??

### Learning objectives: 
Discuss the role of microbial diversity and formation of coupled metabolism in driving global biogeochemical cycles.

#### Specific Questions:
* What are the primary geophysical and biogeochemical processes that create and sustain conditions for life on Earth? How do abiotic versus biotic processes vary with respect to matter and energy transformation and how are they interconnected?  

* Why is Earthâ€™s redox state considered an emergent property?
  

* How do reversible electron transfer reactions give rise to element and nutrient cycles at different ecological scales? What strategies do microbes use to overcome thermodynamic barriers to reversible electron flow?  

* Using information provided in the text, describe how the nitrogen cycle partitions between different redox â€œnichesâ€ and microbial groups. Is there a relationship between the nitrogen cycle and climate change?   

* What is the relationship between microbial diversity and metabolic diversity and how does this relate to the discovery of new protein families from microbial community genomes?  

* On what basis do the authors consider microbes the guardians of metabolism?  



##Data Science Friday 1 - Installation

###Installation Screenshots

![Git Bash](Capture1.JPG)

![RStudio](Capture2.JPG)

![GitHub](Capture3.JPG)

##Data Science Friday 2 - Introduction to Git

###Instructions for portfolio repo creation, initialization and pushing to GitHub

####Configure Git command line to GitHub online account

command: git config --global user.name "Your Name"

command: git config --global user.email "youremail@email.com"

####Locate to "Documents" folder and confirm location

command: cd ~/Documents

command: pwd

####Clone GitHub web MICB425 repository to "Documents"

command: git clone https://github.com/EDUCE-UBC/MICB425 MICB425_materials

####Check MICB425_materials status and pulling 

command: git status

command: git fetch

command: git pull

####Create master repository in "Documents" and create empty .txt file

command: mkdir MICB425_portfolio

command: touch ID.txt

####Initiate and push local repository (after creating online repo on GitHub)

command: git init

command: git add.

command: git commit -m "First commit"

command: git remote add origin https://remote_repository_URL

command: git remote -v

command: git push -u origin master

##Data Science Friday 3 - Pretty_html Challenge

#Pretty_html

### *Zhong Zack Dang (22481148)*
### *version Febuary 22 2018*

* [R Markdown PDF Challenge](#r markdown pdf challenge)
    + [Challenge Goals](##challenge goals)
    + [Here's a header!](###heres a header)
        + [Table Time](###Table Time)

#R Markdown PDF Challenge

The following assignment is an exercise for the reproduction of this .html document using the RStudio and Rmarkdown tools we've shown you in  
class. Hopefully by the end of this, you won't feel at all the way this poor PhD student does. We're here to help, and when it comes to R, the  
internet is a really valuable resource. This open-source program has all kinds of tutorials online.

![](http://www.phdcomics.com/comics/archive/phd011718s.gif)

http://phdcomics.com/ Comic posted 1-17-2018

##Challenge Goals
The goal of this R Markdown html challenge is to give you an opportunity to play with a bunch of different RMarkdown formatting. Consider it a  
chance to flex your RMarkdown muscles. Your goal is to write your own RMarkdown that rebuilds this html document as close to the original as  
possible. So, yes, this means you get to copy my irreverant tone exactly in your own Markdowns. Itâ€™s a little window into my psyche. Enjoy =)

**hint: go to the [PhD Comics website](https://github.com/EDUCE-UBC/MICB425) to see if you can find the image above**  
*If you can't find that exact image, just find a comparable image from the PhD Comics website and include it in your markdown*

###Here's a header!

Letâ€™s be honest, this header is a little arbitrary. But show me that you can reproduce headers with different levels please. This is a level 3 header, for your reference (you can most easily tell this from the table of contents).

####Another header, now with maths

Perhaps you're already confused by the whole markdown thing. Maybe you're so confused you've forgotten how to add. Never fear! ~~A  
calculator~~ R is here:

```{r}
1231521+12341556
```

###Table Time

Or maybe, after youâ€™ve added those numbers, you feel like itâ€™s about time for a table!  
Iâ€™m going to leave all the guts of the coding here so you can see how libraries (R packages) are loaded into R (more on that later). Itâ€™s not terribly  
pretty, but it hints at how R works and how you will use it in the future. The summary function used below is a nice data exploration function that  
you may use in the ^future^.


```{r results = 'asis'}
library(knitr)
kable(summary(cars), caption = "I made this table with kable in the knitr package library")
```

And now youâ€™ve almost finished your first RMarkdown! Feeling excited?  We are! In fact, weâ€™re so excited that maybe we need a big finale eh?  
Hereâ€™s ours! Include a fun gif of your choice!

![](https://media.giphy.com/media/RyXVu4ZW454IM/giphy.gif)

##Data Science Friday 4 - Working with Data

####Exercise 1 - Importing data under correct parameters

```{r}
library(tidyverse)
read.table(file="Saanich.OTU.txt", header=TRUE, row.names=1, sep="\t", na.strings=c("NAN", "NA", "."))
OTU = read.table(file="Saanich.OTU.txt", header=TRUE, row.names=1, sep="\t")
```

####Exercise 2 - Selecting data

```{r}
library(dplyr)

read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t", na.strings=c("NAN", "NA", "."))
metadata = read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t")

filter(metadata, CH4_nM > 100 & Temperature_C < 10) %>%
  select(Depth_m, Temperature_C, CH4_nM)
```

####Exercise 2 - Mutating data

```{r}
library(dplyr)

metadata = read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t")

select(metadata, ends_with("nM")) %>%
  mutate(N2O_uM = N2O_nM/1000) %>%
  mutate(Std_N2O_uM = Std_N2O_nM/1000) %>%
  mutate(CH4_uM = CH4_nM/1000) %>%
  mutate(Std_CH4_uM = Std_CH4_nM/1000) 
```

##Data Science Friday 5 - Plotting Data

```{r tempF, echo=TRUE}
library(tidyverse)
```

###Exercise 1 - Plotting with ggplot

```{r}
library(readr)
library(phyloseq)
library(ggplot2)
library(dplyr)
library(knitr)

ggplot(metadata, aes(x=NH4_uM, y=Depth_m)) +
  geom_point(color="purple", shape=17)
```

###Exercise 2 - Plotting with ggplot and dplyr

```{r}
select(metadata, Temperature_C, Depth_m) %>% 
  mutate(Temperature_F = (Temperature_C * (9/5)+32))

ex2 = select(metadata, Temperature_C, Depth_m) %>% 
  mutate(Temperature_F = (Temperature_C * (9/5)+32))

ggplot(ex2, aes(x=Temperature_F, y=Depth_m)) +
  geom_point()
```

###Exercise 3 - Plotting with ggplot and phyloseq

```{r}
load(file="physeq.RData")
physeq_percent = transform_sample_counts(physeq, function(x) 100 * x/sum(x))
plot_bar(physeq_percent, fill="Order") +
    geom_bar(aes(fill=Order), stat="identity") +
  ggtitle("Saanich Inlet Taxonomic Abundance (10-200m)") +
  xlab("Depth of Sample") 
```

###Exercise 4 - Faceting

```{r}
ex4 = select(metadata, ends_with("uM"), Depth_m)

facet = gather(metadata, key = "Nutrient", value = "uM", ends_with("uM"))

ggplot(facet, aes(x=Depth_m, y=uM))+
  geom_line()+
  geom_point()+
  facet_wrap(~Nutrient, scales="free_y") +
  theme(legend.position="none")
```

##Writing Assignment 1

###Spaceship Earth, A Human-Driven, Microbe-Powered Vessel

####Introduction

Just how dependent is the human race on the global microbiome for our survival and how well can these microbes fare in the absence of humans? In this paper, I will address the statement: "Microbial life can easily live without us; we, however, cannot survive without the global catalysis and environmental transformations it provides." and present arguments as to why I agree with this statement and provide evidence in which to support my case. It is important to delineate the purpose behind this assessment and the implications of the decision to support or refute this statement. Since the birth of humankind, we see the footprints of our species' ability to significantly impact our environment, as evident with the global disappearance of megafauna following over-hunting by early humans (Schrag, 2012). This impact only grew with the invention of agriculture and the later, the Industrial Revolution. We are currently facing a great challenge to ensure the preservation of life on planet Earth as we know it. To accomplish this monumental task, the questions we need to ask in context to the prompt are: How dependent are humans on microbial engines for biogeochemical cycling and environmental transformations? How will microbes respond to these changes and can they fix the problems we have caused? Can us humans prevent the impending catastrophic changes to our planet with our technology alone, or will we need the aid of microbial systems? In this discussion, I will attempt to answer these questions and contemplate our place in the grand scheme of the global ecosystem.

####Microbial Engines in Global Elemental Recycling

In the current biosphere, us humans, and indeed all living organisms, depend on the intricate biogeochemical networks and cycling mediated by microbial communities. As stated by Falkowski et al (2008), the geophysical processes and acid-base chemistry that shaped the abiotic world created niches in which microbes quickly filled, evolving to catalyze the redox reactions that molded the elemental cycles that exist today. Without these microbial systems, it is unlikely that life on Earth, including that of humans, will persist. We will focus on the carbon and the nitrogen cycle to illustrate this point. 

Carbon is an essential element, making up the structures that form the critical molecules of life, including plasma membrane lipids, the sugars in nucleic acids and the backbone structures of amino acids used to make proteins. The most influential role of microbes in the carbon cycle is their fixation of CO2 via photosynthesis. Net primary production of phytoplankton is approximately 50% of the global number, with the remaining 50% achieved by land plants (Longhurst et al, 1995), indicating that microbes are an integral component of the carbon cycle. This also has major implications on the influx of energy into the food web, as heterotrophic microbes and higher eukaryotes depend on his autotrophic transformation of light energy into chemical energy. Also, many organisms directly rely on our oxygen-rich atmosphere for the catabolism of dietary molecules. Besides the explosion of oxygenic photosynthesis by cyanobacteria at around 2.7 Gyr that created the current atmospheric makeup, we also have microbes to thank for the maintenance of the atmospheric oxygen concentration. The marine autotroph Prochlorococcus is likely to be the most abundant genus in the biosphere and accounts for between 13-48% of the global oxygen production (Johnson et al, 2006). Another 30-60% of the global oxygen is produced by single-celled marine algae. Hence we humans, as organisms dependent on oxidative phosphorylation and therefore, oxygen as a terminal electron acceptor for respiration, are at the mercy of microbial carbon fixation and oxygen production through photosynthesis.

Nitrogen is another essential component of life and is a key element in amino acids and nucleic acid bases. As nitrogen gas is inherently inert, catalytically breaking the triple bond between the two N atoms requires a large activation energy barrier to be overcome (Canfield et al, 2010). The nitrogenase enzyme has evolved for exactly that task and quickly spread via horizontal gene transfer between prokaryotes and archaea. Almost all biotic N-fixation into ammonium (NH4+) is dependent on bacteria and archaea in terrestrial and marine ecosystems. This constitutes to around 55% of the total N-fixation, with the remaining nitrogen being fixed by anthropogenic processes such as the Haber-Bosch process and in legume farming. While some eukaryotes can fix nitrogen (and it is only because of symbiotic nitrogenase-expressing bacteria!), this quantity is negligible compared to the aforementioned amount. Further along in the cycle, various bacteria oxidize ammonium into nitrite (NO2-) and nitrate (NO3-) to be used as terminal electron acceptors in respiration. Plants are heavily dependent on these species for their source of nitrogen. On the other end of the cycle, fixed nitrogen must return to the atmosphere as N2 gas. This is accomplished by denitrification and anammox reactions. Anoxic denitrification is highly conserved and is the bacteria and archaea and can also be found in some protozoa and fungi. By contrast, we humans contribute essentially nothing to the return of nitrogen to the atmosphere! We will return to the N-cycle in the final section of this paper, where human implications will be discussed.

####Microbial Resilience

Rough estimates place the grand total of all prokaryotes at 3.8x1030 cells (Whitman, Coleman, Wiebe, 1998). Although there are limitations to the estimation techniques and, with that, a mild degree of error, this number is staggering. The total carbon in prokaryotes can be estimated with more accuracy. At a total of 350-550x1015g of carbon, prokaryotes make up a massive reservoir of organic carbon, between 60-100% of the total plant carbon content (Whitman, Coleman, Wiebe, 1998). Even more impressive is that soil prokaryotes contain cumulatively more nitrogen than all land plants (Whitman, Coleman, Wiebe, 1998). This, once again, asserts the importance of prokaryotes as a nutrient reservoir. In the context of microbial resilience, the sheer microbial abundance in the biosphere is an indicator of stability and persistence. With this huge number of rapidly dividing cells, comes an immense net mutation rate. Whitman, Coleman and Wiebe (1998) speculated that four simultaneous mutations in every gene shared by the marine heterotroph, marine autotroph, soil prokaryote and prokaryotes in domesticated animal populations will occur between every 24 minutes to 170 hours. Thus, with such a high simultaneous mutation rate, genetic diversity within microbial conditions allows for rapid speciation and diversification of metabolic, signaling and stress response pathways. One potent example of this is in antibiotic resistance. Currently, pathogenic bacteria are developing resistance to antibiotics faster than we can discover new, effective ones. In a hospital setting, this can occur within days, indicating the impressive adaptability of microbes through rapid division and spontaneous mutation. 

Not only are prokaryotes experts in environmental adaptation, they also share their abilities with other microorganisms. Horizontal gene transfer is responsible for the vast diversification of metabolic functions across all taxonomic levels. Despite these functions being incredibly widespread and divergent, with some even forming highly niche specific "boutique genes", the "core genes" remain conserved (Falkowski et al, 2008). This can be observed throughout various microbial communities in different environments, where the "core" gene set is always represented by several populations within the community. Unsurprisingly, Falkowski et al (2008) dubbed microbes as the "Guardians of Metabolism". Overall, regardless of the apocalyptic environmental or climate changes that might occur in the future, selective pressures will drive microbes to evolve and distribute new metabolic functions to survive and possibly thrive in a new Earth.

It is clear that microbes will be able to survive just fine without humans, as they have been doing for the most of Earth's geological history. To summarize Nisbet and Sleep's 2001 Nature review in a nutshell, with the creation of new substrates and habitats, ecological niches are formed in which bacteria will inhabit to utilize those substrates. This occurred with the earliest ancestral microorganisms that used sulphate around hydrothermal vents, and later on, with the evolution and explosion of oxygen-producing cyanobacteria, aerobic heterotrophs evolved oxidative phosphorylation to catabolize organic carbon sources. We even see this with anthropogenic-produced molecules. Recently, polychlorinated biphenyl (PCB)-degrading Rhodococcus samples have been cultured and characterized (Leigh et al, 2006). PCBs are a recalcitrant, toxic industrial pollutant that cannot be degraded or removed by conventional means. Another example of microbial resourcefulness is the increase in lignin-degrading bacterial populations downstream of paper production factories. The sheer metabolic resilience and versatility of microorganisms cannot be overstated.

####Humanity's Dependence on and Manipulation of Microbial Systems

With the two examples just mentioned in the previous section, we begin to see how us humans may become dependent on microbes for, not only their geochemical cycling, but also their potential as agents for bioremediation. Our industries and transportation systems are responsible for a titanic amount of CO2 and N2O influx into the atmosphere. We also disrupt biogeochemical cycles by increasing plastic, aluminium, black carbon and concrete burial in global sediments (Mooney, 2016). There is no denial that the carbon cycle has already been thrown out of balance. Current atmospheric CO2 levels are at a record 387 ppm, approaching the boundary limit for irreversible loss of the polar ice caps and severe climate change (Rockstrom et al, 2009). We must engineer a system capable of trapping or fixing CO2 and given the innate capacity for marine autotrophs to fix carbon, it is not far-fetched that such a system will utilize microbes for this purpose. We are also drastically increasing the influx of nitrogen to terrestrial and marine ecosystems. With ever-rising demand for food crops we have increased our nitrogen fertilizer usage by ~800% since the 1960s (Canfield et al, 2010), yet around 60% of this fertilizer is leached away and rained into waterways and oceans, causing massed eutrophication. Interestingly, microbial-mediated denitrification, particularly in anoxic marine habitats has risen sharply and has, for the most part, balanced the anthropogenic nitrogen influx (Canfield et al, 2010). However the capacity of this system is uncertain. Plausible solutions include optimizing fertilizer usage and engineering endosymbiosis of N-fixing bacteria into crops to decrease the need for fertilizers. Hence, with our current levels of CO2 production and nitrogen usage, humans cannot possible continue into the future without the buffering effect of microbial catalyzed biogeochemical cycling of essential elements. Even more likely, is that we have to learn to manipulate these functions in order to maintain our way of life and the preservation of the Earth (Mooney, 2016). 

####Conclusion

In this paper, I have discussed the essential microbial component in biogeochemical cycling of life-sustaining elements including the carbon and nitrogen cycles. I identified that microbial communities, with their high propensity for simultaneous mutations and horizontal gene transfer, as well as sheer abundance, confer a great degree of environmental and metabolic resilience. Lastly, I highlighted the significance of microbial-mediated environmental transformations and metabolic functions in the cushioning of humanity's detrimental effects on the environment and climate. It is for these reasons that I agree with the statement: "Microbial life can easily live without us; we, however, cannot survive without the global catalysis and environmental transformations it provides." In the current state of affairs, we are pushing the planetary boundaries as described by Rockstrom et al (2009) and should we fail to limit our destructive behaviors, it is probable that the Earth will experience severe climate and environmental transformations, along with drastic alterations in existing biogeochemical cycles. The prospect of understanding the global microbial metabolic metagenome and engineering specific functions to suit the planet's needs is an arduous one, possibly involving Petabytes upon Petabytes of metagenomic sequencing data and analysis, not to mention the application and genetic engineering steps. We are headed into the first ever, anthropogenically-engineered eon and it will be our knowledge and manipulation of global microbial functions that lead us to either to human extinction or a lasting, prosperous worldwide symbiosis. 

####References

Canfield, D. E., Glazer, A. N., & Falkowski, P. G. (2010). The Evolution and Future of Earth's
Nitrogen Cycle. Science,330(6001), 192-196. doi:10.1126/science.1186120

Falkowski, P. G., Fenchel, T., & Delong, E. F. (2008). The Microbial Engines That Drive
Earth's Biogeochemical Cycles. Science,320(5879), 1034-1039. doi:10.1126/science.1153213

Johnson, Z. I., Zinser, E. R., Coe, A., McNulty, N. P., Woodward, E. M. S., Chisholm, S. W.
(2006). Niche Partitioning Among Prochlorococcus Ecotypes Along Ocean-Scale Environmental Gradients. Science,311(5768), 1737-1740. doi:10.1126/science.1118052

Kasting, J. F., & Siefert, J. L. (1984). The evolution of the prebiotic atmosphere. Origins of
Life,14(1-4), 75-82. doi:10.1007/bf00933642

Leigh, M. B., Prouzova, P., Mackova, M., Macek, T., Nagle, D. P., & Fletcher, J. S. (2006).
Polychlorinated Biphenyl (PCB)-Degrading Bacteria Associated with Trees in a PCB-Contaminated Site. Applied and Environmental Microbiology,72(4), 2331-2342. doi:10.1128/aem.72.4.2331-2342.2006

Longhurst, A., Sathyendranath, S., Platt, T., & Caverhill, C. (1995). An estimate of global
primary production in the ocean from satellite radiometer data. Journal of Plankton Research,17(6), 1245-1271. doi:10.1093/plankt/17.6.1245

Mooney, C. (2016, January 07). Scientists say humans have now brought on an entirely new
geologic epoch. Retrieved February 16, 2018, from https://www.washingtonpost.com/news/energy-environment/wp/2016/01/07/scientists-say-humans-have-now-brought-on-an-entirely-new-geologic-epoch/

Nisbet, E. G., & Sleep, N. H. (2001). The habitat and nature of early life. Nature,409(6823),
1083-1091. doi:10.1038/35059210

Rockström. J., Steffen. W., Noone. K., Persson. Å., Chapin. F. S., Lambin. E. F., Lenton. T. M.,
Scheffer. M., Folke. C., Schellnhuber H. J.. (2009). A safe operating space for humanity. Nature,461, 472-475

Whitman, W. B., Coleman, D. C., & Wiebe, W. J. (1998). Prokaryotes: The unseen majority.
Proceedings of the National Academy of Sciences,95(12), 6578-6583. doi:10.1073/pnas.95.12.6578


##Module 1 References

#Module 2

###Module 2 Portfolio Check

* Evidence worksheet_04
    + Completion status:
    + Comments:
* Problem Set_03
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments: 

## Problem set_03 ??Metagenomics: Genomic Analysis of Microbial Communities??

### Learning objectives: 
Specific emphasis should be placed on the process used to find the answer. Be as comprehensive as possible e.g. provide URLs for web sources, literature citations, etc.  
*(Reminders for how to format links, etc in RMarkdown are in the RMarkdown Cheat Sheets)*

#### Specific Questions:
* How many prokaryotic divisions have been described and how many have no cultured representatives (microbial dark matter)?
  
2016: 89 bact phyla, 20 archael phyla, determined via small 16S rRNA databases, but there could be up to 1500 bact phyla (microbes in the "shadow biosphere")

* How many metagenome sequencing projects are currently available in the public domain and what types of environments are they sourced from?
  
thousands, 

* What types of on-line resources are available for warehousing and/or analyzing environmental sequence information (provide names, URLS and applications)?  

#####Shotgun metagenomics

Databases - MG-RAST, NCBI-refseq, ING-/M

Assembly - EULER

Binning - S-GSOM

Annotation - KEGG

Analysis pipelines - MEGAN 5

#####Marker gene metagnomics

Standalone Software - OTUbase

Analysis pipelines - SILVA

Denoising - AmpliconNoise

Databases - Ribosomal Database Project (RDP) 

* What is the difference between phylogenetic and functional gene anchors and how can they be used in metagenome analysis?   

* What is metagenomic sequence binning? What types of algorithmic approaches are used to produce sequence bins? What are some risks and opportunities associated with using sequence bins for metabolic reconstruction of uncultivated microorganisms?  

the process of grouping sequences that come from a single genome

types of algorithms: align sequences to database, group to each other base don DNA characteristics (GC content, codon usage)

risks: incomplete coverage of genome sequence (partial data), contamination from different phylogeny

* Is there an alternative to metagenomic shotgun sequencing that can be used to access the metabolic potential of uncultivated microorganisms? What are some risks and opportunities associated with this alternative?  

funcitonal screens (biochemical tests), 3rd generation sequencing (nanopore) good hybrid between single cell sequencing and shotgun sequencing, single cell sequencing, FISH probe, possible to combine methods?

###Module 2 References

#Module 3

###Module 3 Portfolio Check

* Evidence worksheet_05
    + Completion status:
    + Comments:
* Problem set_04
    + Completion status:
    + Comments:
* Writing Assessment_03 
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments:

##Problem Set_04

###Learning objectives:

Gain experience estimating diversity within a hypothetical microbial community

####Part 1: Description and enumeration

Obtain a collection of "microbial" cells from "seawater". The cells were concentrated from different depth intervals by a marine microbiologist travelling along the Line-P transect in the northeast subarctic Pacific Ocean off the coast of Vancouver Island British Columbia.

Sort out and identify different microbial "species" based on shared properties or traits. Record your data in this Rmarkdown using the example data as a guide.

Once you have defined your binning criteria, separate the cells using the sampling bags provided. These operational taxonomic units (OTUs) will be considered separate "species". This problem set is based on content available at What is Biodiversity.

For example, load in the packages you will use.

```{r}
#To make tables
library(kableExtra)
library(knitr)
#To manipulate and plot data
library(tidyverse)
```

Then load in the data. You should use a similar format to record your community data.

```{r}
example_data1 = data.frame(
  number = c(1,2,3),
  name = c("lion", "tiger", "bear"),
  characteristics = c("brown cat", "striped cat", "not a cat"),
  occurences = c(2, 4, 1)
)
```

Finally, use these data to create a table.

```{r}
example_data1 %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)
```

For your sample:

* Construct a table listing each species, its distinguishing characteristics, the name you have given it, and the number of occurrences of the species in the collection.
* Ask yourself if your collection of microbial cells from seawater represents the actual diversity of microorganisms inhabiting waters along the Line-P transect. Were the majority of different species sampled or were many missed?

```{r}
Community_total1 = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20),
  name = c("kiss", "red short string", "red long string", "white spiral", "mike ike", "watermelon", "gummy bear", "marshmallow bear", "lego square", "lego rectangle", "coke", "spider", "big round candy", "skittle", "m&m", "fish", "elongated noodle", "elongated cloud", "elongated oval", "elongated diamond"),
  characteristics = c("teardrop shape", "short string", "long string", "large ellipsoid flat", "rounded rod", "rectangular", "clear bear shape", "opaque bear shape", "large square brick", "small rectangular brick", "coke bottle shape", "spider shape", "sphere", "ellipsoid round", "small elipsoid flat", "fish shape", "short noodle shape", "cloud shape", "oval", "diamond"),
  occurences = c(16,6,10,3,172,1,99,3,3,15,3,6,24,187,243,1,4,3,1,1)
)

Community_total1 %>%
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)

#Many species were missed in the sameple and thus does not represent the true diversity of the community. This is due to the fact that many species were low in abundance and not picked up in the sample. 
```

###Part 2: Collector's Curve

Part 2: Collector's curve
To help answer the questions raised in Part 1, you will conduct a simple but informative analysis that is a standard practice in biodiversity surveys. This analysis involves constructing a collector's curve that plots the cumulative number of species observed along the y-axis and the cumulative number of individuals classified along the x-axis. This curve is an increasing function with a slope that will decrease as more individuals are classified and as fewer species remain to be identified. If sampling stops while the curve is still rapidly increasing then this indicates that sampling is incomplete and many species remain undetected. Alternatively, if the slope of the curve reaches zero (flattens out), sampling is likely more than adequate.

To construct the curve for your samples, choose a cell within the collection at random. This will be your first data point, such that X = 1 and Y = 1. Next, move consistently in any direction to a new cell and record whether it is different from the first. In this step X = 2, but Y may remain 1 or change to 2 if the individual represents a new species. Repeat this process until you have proceeded through all cells in your collection.

For example, we load in these data.

```{r}
example_data2 = data.frame(
  x = c(1,2,3,4,5,6,7,8,9,10),
  y = c(1,2,3,4,4,5,5,5,6,6)
)
```

And then create a plot. We will use a scatterplot (geom_point) to plot the raw data and then add a smoother to see the overall trend of the data.

```{r}
ggplot(example_data2, aes(x=x, y=y)) +
  geom_point() +
  geom_smooth() +
  labs(x="Cumulative number of individuals classified", y="Cumulative number of species observed")
```

For your sample:

* Create a collector's curve.
* Does the curve flatten out? If so, after how many individual cells have been collected?
* What can you conclude from the shape of your collector's curve as to your depth of sampling?


```{r}
Community_sample1 = data.frame(
  x = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20),
  y = c(0,2,6,7,41,41,57,57,57,62,62,64,70,110,164,164,164,164,165,165)
)
```

```{r}
ggplot(Community_sample1, aes(x=x, y=y)) +
  geom_point() +
  geom_smooth() +
  labs(x="Cumulative number of individuals classified", y="Cumulative number of species observed")

#The curve begins to flatten at 8 species and once again at 18 species, the total number of individuals collected is 165.
#Species diversity increases in the sample at the beginning, middle and end of the curve. However, in the intermediary regions, the steepness indicates fewer new species counted.
```

###Part 3: Diversity estimates (alpha diversity)

Using the table from Part 1, calculate species diversity using the following indices or metrics.

####Diversity: Simpson Reciprocal Index
$\frac{1}{D}$ where $D = \sum p_i^2$

$p_i$ = the fractional abundance of the $i^{th}$ species

For example, using the example data 1 with 3 species with 2, 4, and 1 individuals each, D =

species1 = 2/(2+4+1)  species2 = 4/(2+4+1)  species3 = 1/(2+4+1)

1 / (species1^2 + species2^2 + species3^2)

Kiss = 5/147  Red_candy = 1/147  Red_long_candy = 1/147  Mike_ike = 34/147  Gummy_bear = 18/147  Lego_square = 1/147  Lego = 2/147  Spider = 1/147  Big_round = 3/147  Skittle = 31/147  MM = 48/147  Fish = 1/147  cloud = 1/147

Simpsons = 1/ (Kiss^2 + Red_candy^2 + Red_long_candy^2 + Mike_ike^2 + Gummy_bear^2 + Lego_square^2 + Lego^2 + Spider^2 + Big_round^2 + Skittle^2 + MM^2 + Fish^2 + cloud^2)

The higher the value is, the greater the diversity. The maximum value is the number of species in the sample, which occurs when all species contain an equal number of individuals. Because the index reflects the number of species present (richness) and the relative proportions of each species with a community (evenness), this metric is a diveristy metric. Consider that a community can have the same number of species (equal richness) but manifest a skewed distribution in the proportion of each species (unequal evenness), which would result in different diveristy values.

* What is the Simpson Reciprocal Index for your sample?
* What is the chao1 estimate for your original total community?

####Simpson Reciprocal Index (Community)
```{r}
species1 = 16/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species2 = 6/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species3 = 10/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species4 = 3/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species5 = 172/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species6 = 1/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species7 = 99/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species8 = 3/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species9 = 3/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species10 = 15/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species11 = 3/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species12 = 6/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species13 = 24/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species14 = 187/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species15 = 243/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species16 = 1/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species17 = 4/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species18 = 3/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species19 = 1/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species20 = 1/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)

1 / (species1^2 + species2^2 + species3^2 + species4^2 + species5^2 + species6^2 + species7^2 + species8^2 + species9^2 + species10^2 + species11^2 + species12^2 + species13^2 + species14^2 + species15^2 + species16^2 + species17^2 + species18^2 + species19^2 + species20^2)
```

####Simpson Reciprocal Index (Sample)
```{r}
species1 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species2 = 2/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species3 = 4/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species4 = 1/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species5 = 34/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species6 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species7 = 16/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species8 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species9 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species10 = 5/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species11 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species12 = 2/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species13 = 6/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species14 = 30/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species15 = 54/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species16 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species17 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species18 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species19 = 1/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species20 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)

1 / (species1^2 + species2^2 + species3^2 + species4^2 + species5^2 + species6^2 + species7^2 + species8^2 + species9^2 + species10^2 + species11^2 + species12^2 + species13^2 + species14^2 + species15^2 + species16^2 + species17^2 + species18^2 + species19^2 + species20^2)
```

####Richness: Chao1 richness estimator
Another way to calculate diversity is to estimate the number of species that are present in a sample based on the empirical data to give an upper boundary of the richness of a sample. Here, we use the Chao1 richness estimator.

$S_{chao1} = S_{obs} + \frac{a^2}{2b})$

$S_{obs}$ = total number of species observed a = species observed once b = species observed twice or more

So for our previous example community of 3 species with 2, 4, and 1 individuals each, $S_{chao1}$ =

3 + 1^2/(2*2)

Chao1 = (13 + ((6^2)/(2*7)))
* What is the chao1 estimate for your sample?
* What is the chao1 estimate for your original total community?

####Chao1 Richness Estimator (Community)
```{r}
20 + 4^2/(16*2)
```

####Chao1 Richness Estimator (Sample)
```{r}
11 + 2^2/(9*2)
```

###Part 4: Alpha-diversity functions in R
```{r}
library(vegan)

Community_total1_diversity = 
  Community_total1 %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

Community_total1_diversity

diversity(Community_total1_diversity, index="invsimpson")
specpool(Community_total1_diversity)
```

###Part 5: Concluding activity
If you are stuck on some of these final questions, reading the Kunin et al. 2010 and Lundin et al. 2012 papers may provide helpful insights.

* How does the measure of diversity depend on the definition of species in your samples?
* Can you think of alternative ways to cluster or bin your data that might change the observed number of species?
* How might different sequencing technologies influence observed diversity in a sample?